nvflare/private/fed/server/job_runner.py

```diff
@@ -417,6 +420,10 @@ def run(self, fl_ctx: FLContext):
                             else:
                                 deployable_clients = client_sites
 
+                            job = job_manager.get_job(ready_job.job_id, fl_ctx)
+                            if job.meta.get(JobMetaKey.STATUS) != RunStatus.DISPATCHED:
```

Comment:
What do you think if we wrap this two lines as a simple method like
`_check_job_status(job_manager, job_id, job_run_status)`
or add it in job_manager class.

And then here we just call `if _check_job_status(job_manager, ready_job.job_id, RunStatus.DISPATCHED)`

Reply:
changed it.